src/libtorch.cc

```diff
@@ -942,6 +942,16 @@ ModelInstanceState::ReadOutputTensors(
       batchn_shape.push_back(*itr);
     }
 
+    if (batchn_shape.size() == 0) {
+      RESPOND_ALL_AND_RETURN_IF_ERROR(
+          responses, request_count,
+          TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              (std::string("output '") + name +
+               "' is a scalar which is not supported.")
+                  .c_str()));
+    }
+
     responder.ProcessTensor(
         name, output_dtype, batchn_shape, output_buffer,
```

Comment:
By the way, similar to this issue, since there is no way to validate the output shape until the model is executed, when the model "supports" batching, you should verify the first dimension of the output matches the input.

Reply:
Nice chat! I'd like to address that in a follow up ticket+PR though since it will need a custom model to test the same.

Reply:
Yes, this change should go as is, we can use the existing ticket for more general scalar support